src/index.js

```diff
@@ -3,23 +3,33 @@
 var Promise = require('bluebird');
 var sinon   = require('sinon');
 
-var scheduler = function (fn) {
-  process.nextTick(fn);
-};
-
-function schedule (fn) {
-  return function (resolve, reject) {
-    scheduler(function () {
-      fn(resolve, reject);
+function thenable (promiseFactory) {
+  return Object.getOwnPropertyNames(Promise.prototype)
+    .filter(function (method) {
+      return method !== 'then';
+    })
+    .reduce(function (acc, method) {
```

Comment:
Now it makes sense. I knew I was missing something. Everything looks good, sorry for the delay.